samples/socket-api/wasm-src/addr_resolve.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2022 Amazon.com, Inc. or its affiliates. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#ifdef __wasi__
+#include <wasi_socket_ext.h>
+#else
+#include <netdb.h>
+#endif
+
+int
+lookup_host(const char *host)
+{
+    struct addrinfo hints, *res, *result;
+    int errcode;
+    char addrstr[100];
+    void *ptr;
+
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family = AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+
+    errcode = getaddrinfo(host, NULL, &hints, &result);
+    if (errcode != 0) {
+        perror("getaddrinfo");
+        return -1;
+    }
+
+    res = result;
+
+    printf("Host: %s\n", host);
+    while (res) {
+        switch (res->ai_family) {
+            case AF_INET:
+                ptr = &((struct sockaddr_in *)res->ai_addr)->sin_addr;
+                break;
+            case AF_INET6:
+                ptr = &((struct sockaddr_in6 *)res->ai_addr)->sin6_addr;
+                break;
```

Comment:
`break` goes out of the `switch`, but still in `while`. So L46 will use `ptr` in an uninitialized value.

Reply:
yes, it's expected to stay in `while`. The `break` here and in L41 is not expected to break the while loop. `ptr` is only uninitialized if the `ai_family` is set to a value different than `AF_INET` or `AF_INET6`. Having said that, we should have a `default` label here in case the `af_family` has an unexpected value.

Reply:
If `res->ai_family` is neither `AF_INET` nor `AF_INET6`, `ptr` will be some random value, and `inet_ntop` will be failed, and `addrstr` will be some random value, and `printf` may access some random address.

Reply:
yes, that's why I added `default` label to the switch.